src/main/java/network/radicle/tools/github/services/MigrationService.java

```diff
@@ -0,0 +1,94 @@
+package network.radicle.tools.github.services;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.BadRequestException;
+import jakarta.ws.rs.InternalServerErrorException;
+import network.radicle.tools.github.Config;
+import network.radicle.tools.github.clients.IGitHubClient;
+import network.radicle.tools.github.clients.IRadicleClient;
+import network.radicle.tools.github.core.github.Issue;
+import network.radicle.tools.github.core.radicle.actions.CommentAction;
+import network.radicle.tools.github.core.radicle.actions.LifecycleAction;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashSet;
+import java.util.List;
+
+@ApplicationScoped
+public class MigrationService {
+    private static final Logger logger = LoggerFactory.getLogger(MigrationService.class);
+
+    @Inject IGitHubClient github;
+    @Inject IRadicleClient radicle;
+    @Inject Config config;
+
+    public boolean migrateIssues() {
+        var page = 1;
+        var hasMoreIssues = true;
+        var total = 0;
+
+        var partiallyOrNonMigratedIssues = new HashSet<Long>();
+        try {
+            var session = radicle.createSession();
+            logger.info("Created radicle session: {}", session.id);
+            while (hasMoreIssues) {
+                List<Issue> issues = List.of();
+                try {
+                    issues = github.getIssues(page);
+                    var batchSize = issues.size();
+                    logger.debug("Migrating page: {} with size: {}", page, batchSize);
+                    for (var issue : issues) {
+                        //ignore pull requests
+                        if (issue.pullRequest != null) {
+                            continue;
+                        }
+                        total++;
+
+                        var radIssue = issue.toRadicle();
+                        try {
+                            var id = radicle.createIssue(session, radIssue);
+                            if (id != null) {
+                                // update issue's state
+                                radicle.updateIssue(session, id, new LifecycleAction(radIssue.state));
+
+                                // process issue's commends
+                                var commentsPage = 1;
+                                var hasMoreComments = true;
+                                while (hasMoreComments) {
+                                    var comments = github.getComments(issue.number, commentsPage);
+                                    for (var comment : comments) {
+                                        radicle.updateIssue(session, id, new CommentAction(comment.body));
```

Comment:
the comment body should probably be altered to contain author information (and possibly any other metadata available in GitHub that cannot be represented in a straightforward way in Radicle)